metric_providers/network/proxy/proxy_provider.py

```diff
@@ -0,0 +1,146 @@
+# pylint: disable=no-member,consider-using-with,subprocess-popen-preexec-fn,import-error,too-many-instance-attributes,too-many-arguments
+
+# This code handles the setup of the proxy we use to monitor the network connections in the docker containers.
+# Structurally it is a copy of the BaseMetricProvider but because we need to do things slightly different it is a copy.
+# In the future this might be implemented as a proper provider.
+
+import os
+from pathlib import Path
+import subprocess
+import signal
+import sys
+import time
+import re
+from datetime import datetime, timezone
+import platform
+import subprocess
+from packaging.version import parse
+
+from db import DB
+from global_config import GlobalConfig
+from metric_providers.base import MetricProviderConfigurationError
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+
+class NetworkConnectionsProxyContainerProvider:
+
+    def __init__(self, host_ip=None):
+        self._metric_name = 'dockerproxy'
+        self._tmp_folder = '/tmp/green-metrics-tool'
+        self._conf_file = f"{CURRENT_DIR}/proxy_conf.conf"
+        self._filename = f"{self._tmp_folder}/proxy.log"
+        self._ps = None
+        self._host_ip = host_ip
+        self._has_started = False
+
+        Path(self._tmp_folder).mkdir(exist_ok=True)
+
+    # This needs to be static as we want to check the system before we initialise all the providers
+    def check_system(self):
+
+        output = subprocess.check_output(["tinyproxy", "-v"], stderr=subprocess.STDOUT, text=True)
+        version_string = output.strip().split()[1].split('-')[0]
+        if parse(version_string) >= parse("1.11"):
+            return True
+
+        raise MetricProviderConfigurationError('Tinyproxy needs to be version 1.11 or greater.')
+
+    def has_started(self):
```

Comment:
why are these functions has_started and get_stderr reimplemented with the same code from the base.py here?

I asked the question about inheritance earlier, so I guess this is related